api.go

```diff
@@ -1727,3 +1727,678 @@ func (sc *Client) GetTimeseries(corpName, siteName string, query url.Values) ([]
+	var responseBody ResponseSiteRedactionBodyList
+	err := json.Unmarshal(response, &responseBody)
+	if err != nil {
+		return ResponseSiteRedactionBodyList{}, err
+	}
+	return responseBody, nil
+}
+
+func getResponseSiteRedactionBody(response []byte) (ResponseSiteRedactionBody, error) {
+	var responseBody ResponseSiteRedactionBody
+	err := json.Unmarshal(response, &responseBody)
+	if err != nil {
+		return ResponseSiteRedactionBody{}, err
+	}
+	return responseBody, nil
+}
+
+//GetSiteRedactionByID get a site redaction by id
+func (sc Client) GetSiteRedactionByID(corpName, siteName, id string) (ResponseSiteRedactionBody, error) {
+	resp, err := sc.doRequest("GET", fmt.Sprintf("/v0/corps/%s/sites/%s/redactions/%s", corpName, siteName, id), "")
+	if err != nil {
+		return ResponseSiteRedactionBody{}, err
+	}
+	return getResponseSiteRedactionBody(resp)
+}
+
+// UpdateSiteRedactionByID updates a site redaction and returns a response
+func (sc Client) UpdateSiteRedactionByID(corpName, siteName string, id string, body CreateSiteRedactionBody) (ResponseSiteRedactionBody, error) {
+	b, err := json.Marshal(body)
+	if err != nil {
+		return ResponseSiteRedactionBody{}, logError(1, fmt.Errorf("%s with input %#v", err.Error(), body))
+	}
+	resp, err := sc.doRequest("PATCH", fmt.Sprintf("/v0/corps/%s/sites/%s/redactions/%s", corpName, siteName, id), string(b))
+	if err != nil {
+		return ResponseSiteRedactionBody{}, logError(1, fmt.Errorf("%s with input %v", err.Error(), string(b)))
+	}
+	return getResponseSiteRedactionBody(resp)
+}
+
+// DeleteSiteRedactionByID deletes a redaction and returns an error
+func (sc Client) DeleteSiteRedactionByID(corpName, siteName string, id string) error {
+	_, err := sc.doRequest("DELETE", fmt.Sprintf("/v0/corps/%s/sites/%s/redactions/%s", corpName, siteName, id), "")
+	if err != nil {
+		return logError(1, fmt.Errorf("%s could not delete %s in %s in %s", err.Error(), id, corpName, siteName))
+	}
+	return nil
+}
+
+// GetAllSiteRedactions Lists the Sites Redactions
+func (sc *Client) GetAllSiteRedactions(corpName, siteName string) (ResponseSiteRedactionBodyList, error) {
+	resp, err := sc.doRequest("GET", fmt.Sprintf("/v0/corps/%s/sites/%s/redactions", corpName, siteName), "")
+
+	if err != nil {
+		return ResponseSiteRedactionBodyList{}, logError(1, fmt.Errorf("%s with %s %s ", err.Error(), corpName, siteName))
+	}
+
+	var responseRulesList ResponseSiteRuleBodyList
+	err = json.Unmarshal(resp, &responseRulesList)
+	if err != nil {
+		return ResponseSiteRedactionBodyList{}, err
+	}
+
+	return getResponseSiteRedactionListBody(resp)
+}
+
+//CreateCorpRuleBody contains the rule of a Corp
+type CreateCorpRuleBody struct {
+	SiteNames     []string    `json:"siteNames,omitempty"`      //Sites with the rule available. Rules with a global corpScope will return '[]'.
+	Type          string      `json:"type,omitempty,omitempty"` //(group, single)
```

Comment:
Type is `request|signal`
Also, there are no rate limit corp rules, so don't need to worry about that here.